airflow/api_fastapi/core_api/datamodels/dag_run.py

```diff
@@ -65,3 +65,14 @@ class DAGRunResponse(BaseModel):
     triggered_by: DagRunTriggeredByType
     conf: dict
     note: str | None
+
+
+class TriggerDAGRunPostBody(BaseModel):
+    """Trigger DAG Run Serializer for POST body."""
+
+    dag_run_id: str | None
+    logical_date: datetime
```

Comment:
According to legacy endpoint implementation, logical_date can't be None. So, I had to make this None.

Reply:
I think we want to remove this entirely and let the backend interpret the logical_date. I assume that has to happen at a deeper layer than use `fastapi` is that right @uranusjr ?

Reply:
I referred to the implementation and updated this PR

Reply:
Yeah a deeper layer should infer logical date from the data interval, or use `now()` if those are not provided. This should not take the argument at all.